src/platforms/android/common/performance/instrumentation/apollo.mdx

```diff
@@ -0,0 +1,116 @@
+---
+title: Apollo Integration
+sidebar_order: 20
+description: "Learn how to capture the performance of Apollo GraphQL client."
+---
+
+<Note>
+
+Capturing transactions requires that you first <PlatformLink to="/performance/">set up performance monitoring</PlatformLink> if you haven't already.
+
+</Note>
+
+Sentry Apollo integration provides the `SentryApolloInterceptor`, which creates a span for each outgoing HTTP request executed with an [Apollo Android](https://github.com/apollographql/apollo-android) GraphQL client.
+
+## Install
+
+```xml {tabTitle:Maven}
+<dependency>
+    <groupId>io.sentry</groupId>
+    <artifactId>sentry-apollo</artifactId>
+    <version>{{ packages.version('sentry.java.apollo', '5.2.0') }}</version>
+</dependency>
+```
+
+```groovy {tabTitle:Gradle}
+implementation 'io.sentry:sentry-apollo:{{ packages.version('sentry.java.apollo', '5.2.0') }}'
+```
+
+For other dependency managers see the [central Maven repository](https://search.maven.org/artifact/io.sentry/sentry-apollo).
+
+## Configure
+
+Add `SentryApolloInterceptor` to Apollo builder:
+
+```java
+import com.apollographql.apollo.ApolloClient;
+import io.sentry.apollo.SentryApolloInterceptor;
+
+ApolloClient apollo = ApolloClient.builder()
+    .serverUrl("https://your-api-host/")
+    .addApplicationInterceptor(new SentryApolloInterceptor())
+    .build();
+```
+
+```kotlin
+import com.apollographql.apollo.ApolloClient
+import io.sentry.apollo.SentryApolloInterceptor
+
+val apollo = ApolloClient.builder()
+    .serverUrl("https://your-api-host/")
+    .addApplicationInterceptor(SentryApolloInterceptor())
+    .build()
+```
+
+## Using With Kotlin Coroutines
+
+To make sure that a coroutine has access to the correct Sentry context, an instance of `SentryContext` must be provided when launching a coroutine:
+
+```kotlin
+import io.sentry.kotlin.SentryContext
+import com.apollographql.apollo.exception.ApolloException
+import kotlinx.coroutines.launch
+
+launch(SentryContext()) {
+  val response = try {
+    apollo.query(..).toDeferred().await()
+  } catch (e: ApolloException) {
+    // handle protocol errors
+    return@launch
+  }
+}
+```
```

Comment:
should we avoid that for Android too? since its single hub, no need to have a SentryContext